tests/integration/kubernetes/k8s-guest-pull-image-signature.bats

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/env bats
+# Copyright (c) 2024 IBM Corporation
+# Copyright (c) 2024 Alibaba Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+load "${BATS_TEST_DIRNAME}/lib.sh"
+load "${BATS_TEST_DIRNAME}/confidential_common.sh"
+load "${BATS_TEST_DIRNAME}/confidential_kbs.sh"
+
+export KBS="${KBS:-false}"
+
+setup() {
+    if ! is_confidential_runtime_class; then
+        skip "Test not supported for ${KATA_HYPERVISOR}."
+    fi
+
+    [ "${SNAPSHOTTER:-}" = "nydus" ] || skip "None snapshotter was found but this test requires one"
+
+    setup_common
+    UNSIGNED_UNPROTECTED_REGISTRY_IMAGE="quay.io/prometheus/busybox:latest"
+    UNSIGNED_PROTECTED_REGISTRY_IMAGE="ghcr.io/confidential-containers/test-container-image-rs:unsigned"
+    COSIGN_SIGNED_PROTECTED_REGISTRY_IMAGE="ghcr.io/confidential-containers/test-container-image-rs:cosign-signed"
+    COSIGNED_SIGNED_PROTECTED_REGISTRY_WRONG_KEY_IMAGE="ghcr.io/confidential-containers/test-container-image-rs:cosign-signed-key2"
+}
+
+function setup_kbs_image_policy() {
+    if [ "${KBS}" = "false" ]; then
+        skip "Test skipped as KBS not setup"
+    fi
+
+    default_policy="${1:-insecureAcceptAnything}"
+
+    policy_json=$(echo "{
+    \"default\": [
+        {
+            \"type\": \"${default_policy}\"
+        }
+    ],
+    \"transports\": {
+        \"docker\": {
+            \"ghcr.io/confidential-containers/test-container-image-rs\": [
+                {
+                    \"type\": \"sigstoreSigned\",
+                    \"keyPath\": \"kbs:///default/cosign-public-key/test\"
+                }
+            ],
+            \"quay.io/prometheus\": [
+                {
+                    \"type\": \"insecureAcceptAnything\"
+                }
+            ]
+        }
+    }
+}")
+
+    public_key=$(echo "-----BEGIN PUBLIC KEY-----
+MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEwQEjdCiL3ILUf07NDkDVhgKCj1C6
+BsCfmM/zt1kNSj0/+nAqA+25XfyClYq2lJFJ6TkgCsf57cTCkXYDz9c+Yg==
+-----END PUBLIC KEY-----")
+
+    if ! is_confidential_hardware; then
+        kbs_set_allow_all_resources
+    fi
+
+    kbs_set_resource "default" "security-policy" "test" "${policy_json}"
+    kbs_set_resource "default" "cosign-public-key" "test" "${public_key}"
+}
+
+function create_pod_yaml_with_signed_image() {
```

Comment:
The name is slightly misleading since this function will only enable the signature verification if `${2}` is `true` or empty. And the two first tests don't seem to pass signed images.

Also, this is duplicating a fair amount of code already present in `create_pod_yaml_with_encrypted_image()` and `create_pod_yaml_with_private_image()`. Please consider factoring out some common helper.

Reply:
Yup, I added a commit to do the refactoring